mpf/devices/segment_display/segment_display_text.py

```diff
@@ -76,28 +76,36 @@ def _create_characters(cls, text: str, display_size: int, collapse_dots: bool, c
         """Create characters from text and color them.
 
         - Colors are used from the left to the right (starting with the first character).
-        - If colors are shorter than text the last color is repeated for text.
-        - The first color is used to pad the text to the left if text is shorter than the display - thus text is right
-          aligned.
         - Dots and commas are embedded on the fly.
+        - Text will be right aligned on the display, thus if text is shorter than display spaces will be padded before the text
+        - Provided colors are assumed to be for the text itself, thus the list of colors is "right aligned" too
+        - If list of colors is less than the display size the list will be extended with white as default color on the left endswith
+          so that the provided colors are for the text and not being used for invisible spaces
         """
         char_list = []
-        left_pad_color = colors[0] if colors else None
-        default_right_color = colors[len(colors) - 1] if colors else None
         uncolored_chars = cls._embed_dots_and_commas(text, collapse_dots, collapse_commas, use_dots_for_commas)
-        colors = colors[-len(uncolored_chars):]
-        for char_code, char_has_dot, char_has_comma in uncolored_chars:
-            color = colors.pop(0) if colors else default_right_color
-            char_list.append(DisplayCharacter(char_code, char_has_dot, char_has_comma, color))
 
-        # ensure list is the same size as the segment display (cut off on left or right justify characters)
-        current_length = len(char_list)
+        # Adujust the color array if needed
+        color_length = len(colors)
+        if color_length > display_size:
```

Comment:
How much do we really want to handle here? I think it is reasonable for a designer to send one color (for all segments) or a list of colors (one for each segment), and those are the two options. 

It doesn't seem reasonable to expect MPF to "correctly" handle coloring a seven segment display when you send it four colors or something 🤷. I think we should strip down all this color-filling logic and throw an exception if the list of colors is neither 1 nor the display size. Unless I'm missing a use case?

Reply:
Very true, that is what I asked myself as well. I tried to be minimal invasive with my code changes, so that is what I came up with. But it seems we have the same understanding here and thus I am happy to change the color handling. Will provide a new PR in the next days including all the other feedback you gave.